misc/python/materialize/xcompile.py

```diff
@@ -53,6 +53,31 @@ def target(arch: Arch) -> str:
     return f"{arch}-unknown-linux-gnu"
 
 
+def target_cpu(arch: Arch) -> str:
+    """Return the CPU micro architecture, assuming a Linux target, we should use for Rust compilation."""
+    if arch == Arch.X86_64:
+        return "x86-64-v3"
+    elif arch == Arch.AARCH64:
+        return "neoverse-n1"
+    else:
+        raise RuntimeError("unreachable")
+
+
+def target_features(arch: Arch) -> list[str]:
+    """
+    Returns a list of CPU features we should enable for Rust compilation.
+
+    Note: We also specify the CPU target when compiling Rust which should enable the majority of
+    available CPU features.
+    """
+    if arch == Arch.X86_64:
+        return ["+aes"]
+    elif arch == Arch.AARCH64:
+        return ["+aes", "+sha2"]
+    else:
+        raise RuntimeError("unreachable")
```

Comment:
Probably worth dropping a note in here indicating that the target CPU and target features should be kept in sync with `bin/ci-builder`, and vice-versa.

Reply:
Done!